### PR TITLE
module calls as objects

### PR DESCRIPTION
**Pull request by jabielecki**
## Description

Add "module calls as objects" section. Behavior seen on tf between 0.13.6 and the latest 1.0.1.

Also copyedit.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -175,109 +175,109 @@ All code should support the latest minor version of the latest Terraform GA rele
...
+      security_groups   = lookup(each.value, "security_groups", null)
+      source_dest_check = lookup(each.value, "source_dest_check", false)
+      description       = lookup(each.value, "description", null)
+      attachment {
+        instance     = aws_instance.this[each.value.fw_name].id
+        device_index = lookup(each.value, "index", null)
+      }
+      tags = { "Name" = each.key }
+    }
+    ```
```
Comment (jabielecki) [inline README.md:278]: I did no changes here, I've just indented the whole code.

File: README.md [hunk 2]
```diff
@@ -320,6 +320,77 @@ Input variable structure as in good example 2 is preferred over example 1 where
...
+
+    Terraform will perform the following actions:
+
+      # random_pet.main will be destroyed
+      - resource "random_pet" "main" {
+          - id        = "dynamic-basilisk-evident-bat" -> null
+          - length    = 2 -> null
+          - prefix    = "dynamic-basilisk" -> null
+          - separator = "-" -> null
+        }
```
Comment (jabielecki) [inline README.md:373]: The crux of the problem. This entire section is about how to avoid this deletion.

**Review by damianfedeczko — APPROVED** [review]
Good lesson, thanks!

**Review by devsecfranklin — APPROVED** [review]
LGTM this is good housekeeping!

**Pull request merged**